dm/quick-start-with-dm.md

```diff
@@ -1,182 +1,452 @@
+[TiDB Data Migration (DM)](/dm/dm-overview.md) is a powerful tool that replicates data from MySQL-compatible databases to TiDB. This guide shows you how to set up a local TiDB DM environment for development and testing using [TiUP Playground](/tiup/tiup-playground.md).
 
 > **Note:**
 >
-> If your target machine's operating system supports SELinux, ensure that SELinux is **disabled**.
+> For production deployments, see [Deploy a DM Cluster Using TiUP](/dm/deploy-a-dm-cluster-using-tiup.md).
 
-## Step 1: Deploy a DM cluster
+## Step 1: Set up the test environment
 
-1. Install TiUP, and install [`dmctl`](/dm/dmctl-introduction.md) using TiUP:
+[TiUP](/tiup/tiup-overview.md) is a cluster operation and maintenance tool. Its Playground feature lets you quickly launch a temporary local environment with a TiDB database and TiDB DM for development and testing.
 
-    {{< copyable "shell-regular" >}}
+1. Install TiUP:
 
     ```shell
     curl --proto '=https' --tlsv1.2 -sSf https://tiup-mirrors.pingcap.com/install.sh | sh
-    tiup install dm dmctl
     ```
 
-2. Generate the minimal deployment topology file of a DM cluster:
+    > **Note:**
+    >
+    > If you have an existing installation of TiUP, ensure it is updated to v1.16.1 or later to use the `--dm-master` and `--dm-worker` flags. To check your current version:
+    >
+    > ```shell
+    > tiup --version
+    > ```
+    >
+    > To upgrade TiUP to the latest version, run:
+    >
+    > ```shell
+    > tiup update --self
+    > ```
 
-    {{< copyable "shell-regular" >}}
+2. Start TiUP Playground with a TiDB target database and DM components:
 
+    ```shell
+    tiup playground --dm-master 1 --dm-worker 1 --tiflash 0 --without-monitor
     ```
-    tiup dm template
+
+3. Verify the environment by checking in the output whether TiDB and DM are running:
+
+    ```text
+    TiDB Playground Cluster is started, enjoy!
+
+    Connect TiDB:    mysql --host 127.0.0.1 --port 4000 -u root
+    Connect DM:      tiup dmctl --master-addr 127.0.0.1:8261
+    TiDB Dashboard:  http://127.0.0.1:2379/dashboard
     ```
 
-3. Copy the configuration information in the output, and save it as the `topology.yaml` file with the modified IP address. Deploy the DM cluster with the `topology.yaml` file using TiUP:
+4. Keep `tiup playground` running in the current terminal and open a new terminal for the following steps.
+
+    This playground environment provides the running processes for the target TiDB database and the replication engine (DM-master and DM-worker). It will handle the data flow: MySQL (source) → DM (replication engine) → TiDB (target).
 
-    {{< copyable "shell-regular" >}}
+## Step 2: Prepare a source database (optional)
+
+You can use one or multiple MySQL instances as a source database. If you do not have a MySQL-compatible instance already, you can create one for testing purposes follow these steps. If you already have a source MySQL database, skip to the Step 3.
+
+<SimpleTab>
```

Comment:
```suggestion
<SimpleTab groupId="os">
```
Adding `groupId="os"` to the SimpleTab component ensures proper tab group synchronization. This parameter links related tabs together, allowing them to switch in unison when the operating system selection changes.